static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx

```diff
@@ -27,12 +27,83 @@ import {Organization} from 'sentry/types';
 import {trackAnalytics} from 'sentry/utils/analytics';
 import {SourceMapWizardBlueThunderAnalyticsParams} from 'sentry/utils/analytics/stackTraceAnalyticsEvents';
 
+const SOURCE_MAP_SCRAPING_REASON_MAP = {
+  not_found: {
+    shortName: t('Not Found'),
+    explanation: t('The source map could not be found at its defined location.'),
+  },
+  disabled: {
+    shortName: t('Disabled'),
+    explanation: t('JavaScript source fetching is disabled in your project settings.'),
+  },
+  invalid_host: {
+    shortName: t('Invalid Host'),
+    explanation: t(
+      'The source map location was not in the list of allowed domains in your project settings, or the URL was malformed.'
+    ),
+  },
+  permission_denied: {
+    shortName: t('Permission Denied'),
+    explanation: t(
+      'Permission to access the source map was denied by the server hosting the source map.'
+    ),
+  },
+  timeout: {
+    shortName: t('Timeout'),
+    explanation: t('The request to download the source map timed out.'),
+  },
+  download_error: {
+    shortName: t('Download Error'),
+    explanation: t('There was an error while downloading the source map.'),
+  },
+  other: {
+    shortName: t('Unknown'),
+    explanation: t('Fetching the source map failed for an unknown reason.'),
+  },
+} as const;
+
+const SOURCE_FILE_SCRAPING_REASON_MAP = {
+  not_found: {
+    shortName: t('Not Found'),
+    explanation: t('The source file could not be found at its defined location.'),
+  },
+  disabled: {
+    shortName: t('Disabled'),
+    explanation: t('JavaScript source fetching is disabled in your project settings.'),
+  },
+  invalid_host: {
+    shortName: t('Invalid Host'),
+    explanation: t(
+      'The source file location was not in the list of allowed domains in your project settings, or the URL was malformed.'
+    ),
+  },
+  permission_denied: {
+    shortName: t('Permission Denied'),
+    explanation: t(
+      'Permission to access the source file was denied by the server hosting it.'
```

Comment:
l: I'm wondering if it makes sense to suggest common reasons why some of these failure points happen. For example, with `permission_denied`, it might be, because the error happened behind a login-protected page which the scraper couldn't access, right? 

Just an idea. I can see that this list could become huge, depending on the failure reason. Definitely not a blocker for this PR.

Reply:
Good feedback! I'll try to find out what the common causes for these cases are and add some text!